libs/gl-client/src/signer/mod.rs

```diff
@@ -713,14 +806,97 @@ impl Signer {
     /// Create a Node stub from this instance of the signer, configured to
     /// talk to the corresponding node.
     pub async fn node(&self) -> Result<Client, anyhow::Error> {
-        node::Node::new(self.node_id(), self.network, self.tls.clone())
+        node::Node::new(self.node_id(), self.network, self.creds.clone())?
             .schedule()
             .await
     }
 
     pub fn version(&self) -> &'static str {
         VERSION
     }
+
+    /// Creates a base64 string called a rune which is used to authorize
+    /// commands on the node and to issue signatures from the signer. Each new
+    /// rune must contain a `pubkey` field that equals the public key that is
+    /// used to sign-off signature requests. Nobody can remove restrictions from
+    /// a rune.
+    ///
+    /// If a `rune` is supplied the restrictions are added to this rune. This
+    /// way one can invoke a rune that only allows for a subset of commands.
+    ///
+    /// `restrictions` is a vector of restrictions where each restriction itself
+    /// is a vector of one ore more alternatives.
+    ///
+    /// - =: passes if equal ie. identical. e.g. method=withdraw
+    /// - /: not equals, e.g. method/withdraw
+    /// - ^: starts with, e.g. id^024b9a1fa8e006f1e3937f
+    /// - $: ends with, e.g. id$381df1cc449605.
+    /// - ~: contains, e.g. id~006f1e3937f65f66c40.
+    /// - <: is a decimal integer, and is less than. e.g. time<1656759180
+    /// - \>: is a decimal integer, and is greater than. e.g. time>1656759180
+    /// - {: preceeds in alphabetical order (or matches but is shorter), e.g. id{02ff.
+    /// - }: follows in alphabetical order (or matches but is longer), e.g. id}02ff.
+    /// - #: a comment, ignored, e.g. dumb example#.
+    /// - !: only passes if the name does not exist. e.g. something!.  Every other operator except # fails if name does not exist!
+    ///
+    /// # Examples
+    /// This creates a fresh rune that is only restricted to a pubkey:
+    ///
+    /// `create_rune(None, vec![vec!["pubkey=000000"]])`
+    ///
+    /// "wjEjvKoFJToMLBv4QVbJpSbMoGFlnYVxs8yy40PIBgs9MC1nbDAmcHVia2V5PTAwMDAwMA"
+    ///
+    /// This adds a restriction to the rune, in this case a restriction that only
+    /// allows to call methods that start with "list" or "get", basically a
+    /// read-only rune:
+    ///
+    /// `create_rune("wjEjvKoFJToMLBv4QVbJpSbMoGFlnYVxs8yy40PIBgs9MC1nbDAmcHVia2V5PTAwMDAwMA", vec![vec!["method^list", "method^get"]])`
+    ///
+    pub fn create_rune(
+        &self,
+        rune: Option<&str>,
+        restrictions: Vec<Vec<&str>>,
+    ) -> Result<String, anyhow::Error> {
+        if let Some(rune) = rune {
+            // We got a rune, add restrictions to it!
+            let mut rune: Rune = Rune::from_base64(rune)?;
```

Comment:
It might be useful to do validity checks (like checking for pubkey field) here too since it doesn't look like [from_base64](https://github.com/nepet/futhark/blob/e659f4030a116867e1ecbb124342782b1f71cfbd/src/lib.rs#L707) checks for specific things. Maybe not a big deal but I'm thinking if there are more required restrictions it might be useful to consolidate the validity checks in a function that's called for both creating and modifying runes

Reply:
I initially implemented this check, but later removed it to avoid being patronizing towards developers who may want to create runes for their own use cases.